src/models/ComponentStyle.js

```diff
@@ -36,15 +34,17 @@ export default (nameGenerator: NameGenerator) => {
      * Returns the hash to be injected on render()
      * */
     generateAndInjectStyles(executionContext: Object) {
-      const flatCSS = flatten(this.rules, executionContext).join('')
-        .replace(/^\s*\/\/.*$/gm, '') // replace JS comments
-      const hash = hashStr(this.componentId + flatCSS)
+      const flatCSS = flatten(this.rules, executionContext)
+      const hash = hashStr(this.componentId + flatCSS.join(''))
+
```

Comment:
@theKashey I'm not quite sure what you're referring to? This is no change to the logic before :) we're relying on the fact that the componentid plus the hash are unique